core/extensions/impl/crypto_extension.cpp

```diff
@@ -54,6 +57,32 @@ namespace kagome::extensions {
         : kVerifyFail;
   }
 
+  runtime::SizeType CryptoExtension::ext_sr25519_verify(
+      runtime::WasmPointer msg_data, runtime::SizeType msg_len,
+      runtime::WasmPointer sig_data, runtime::WasmPointer pubkey_data) {
+    // for some reason, 0 and 5 are used in the reference implementation, so
+    // it's better to stick to them in ours, at least for now
+    static constexpr uint32_t kVerifySuccess = 0;
+    static constexpr uint32_t kVerifyFail = 5;
+
+    const auto &msg = memory_->loadN(msg_data, msg_len);
+    auto sig_bytes =
+        memory_->loadN(sig_data, SR25519_SIGNATURE_SIZE).toVector();
+    uint8_t signature[SR25519_SIGNATURE_SIZE];
+    gsl::span<uint8_t> sig_span(signature);
+    std::copy(sig_bytes.begin(), sig_bytes.end(), sig_span.begin());
```

Comment:
what type is `sig_bytes`? 
you could use it directly, no need to make a copy (`std::copy` into `uint8_t signature[]`)